src/vm/scanning.rs

```diff
@@ -203,20 +195,12 @@ pub trait Scanning<VM: VMBinding> {
     /// * `tls`: The GC thread that is performing the thread scan.
     fn notify_initial_thread_scan_complete(partial_scan: bool, tls: VMWorkerThread);
 
-    /// Scan all the mutators for roots.
-    ///
-    /// The `memory_manager::is_mmtk_object` function can be used in this function if
-    /// -   the "is_mmtk_object" feature is enabled.
-    ///
-    /// Arguments:
-    /// * `tls`: The GC thread that is performing this scanning.
-    /// * `factory`: The VM uses it to create work packets for scanning roots.
-    fn scan_roots_in_all_mutator_threads(
-        tls: VMWorkerThread,
-        factory: impl RootsWorkFactory<VM::VMEdge>,
-    );
-
-    /// Scan one mutator for roots.
+    /// Scan one mutator for stack roots. MMTk assumes a binding is able to scan
+    /// any given thread for its stack roots. If a binding cannot scan the stack for
+    /// a given thread, it can leave this method empty, and deal with stack
```

Comment:
The word "given" (as an adjective) is a bit ambiguous.  In mathematics, "given" refers to conditions that are known to be true.  So "a binding cannot scan the stack for a given thread" can be interpreted as "there exists one thread in the VM that the binding cannot scan (but it can scan other threads)".  The binding must be able to scan all threads (stacks). I think this comment is contrasting two cases: (1) Whenever MMTk gives the binding a thread reference, the binding can scan it; and (2) The binding can only enumerate threads by itself, and scan them while enumerating.

```suggestion
    /// Scan one mutator for stack roots.
    ///
    /// Some VM bindings may not be able to implement this method.
    /// For example, the VM binding may only be able to enumerate all threads and
    /// scan them while enumerating, but cannot scan stacks individually when given
    /// the references of threads.
    /// In that case, it can leave this method empty, and deal with stack
```

Reply:
Sounds good.